source/js/apiHelpers.js

```diff
@@ -44,27 +42,28 @@ export async function getDetailedRecipeInfoBulk(idsToFetch) {
 }
 
 /**
- * Helper function to extract recipe ids from complex search
+ * Helper function to extract recipe ids from complex search ONLY
  * @param {Object} - list of recipes search results from complex search
- * @returns {Object} - [Recipes IDs to fetch, Recipe ids already in local storage]
+ * @returns {Object} - [Recipes IDs to fetch, Recipe JSONs already in local storage]
  */
 export function extractIDs(data) {
   const { results } = data;
   if (!results) {
     return [];
   }
   const idsToFetch = [];
-  const idsInLocalStorage = [];
+  const recipesInLocalStorage = [];
   const recipeData = getAllRecipes();
   results.forEach((result) => {
     const { id } = result;
     if (recipeData[id] == undefined) {
       idsToFetch.push(id);
     } else {
-      idsInLocalStorage.push(id);
+      recipesInLocalStorage.push(recipeData[id]);
     }
   });
-  return [idsToFetch, idsInLocalStorage];
+  console.log(idsToFetch, recipesInLocalStorage);
+  return [idsToFetch, recipesInLocalStorage];
```

Comment:
not completely sure how you're using this function so idk if this was intentional but i just wanted to emphasize that you're currently returning a list where both elements are lists. if you want to combine all of the elements instead, you can use the spread operator in js like this 
<img width="250" alt="Screen Shot 2021-11-22 at 12 16 23 PM" src="https://user-images.githubusercontent.com/33165426/142929061-4c153873-766c-4bf5-91c4-94bf85f3fed8.png">

Reply:
yea so Im trying to separate the ids that I need to fetch from API and the ids that are already found in local storage. I need them separate to call the API with idsToFetch and then recipesInLocalStorage will be the JSONs already found in local storage. 

Thats why I have two lists within a list. Is there a cleaner way to do this?